boot/seal.go

```diff
@@ -134,91 +123,74 @@ func sealKeyToModeenvImpl(key, saveKey secboot.BootstrappedContainer, model *ass
 		}
 	}
 
+	method := device.SealingMethodTPM
 	if flags.HasFDESetupHook {
-		return sealKeyToModeenvUsingFDESetupHook(key, saveKey, model, modeenv, flags)
+		method = device.SealingMethodFDESetupHook
+	} else {
+		if flags.StateUnlocker != nil {
+			relock := flags.StateUnlocker()
+			defer relock()
+		}
 	}
-
-	if flags.StateUnlocker != nil {
-		relock := flags.StateUnlocker()
-		defer relock()
+	if err := sealKeyToModeenvForMethod(method, key, saveKey, model, modeenv, flags); err != nil {
+		return err
 	}
-	return sealKeyToModeenvUsingSecboot(key, saveKey, model, modeenv, flags)
+	return nil
```

Comment:
nitpick: is there any reason for not doing directly `return sealKeyToModeenvUsingSecboot(...)`?

Reply:
Fixed.